### ISSUE TITLE
Implement breached password check when resetting or activating accounts

### ISSUE THREAD

**Issue by tomudding** [GEWIS/gewisweb#1330]
A validator for this was introduced in `laminas/laminas-validator`, see [undisclosed passwords validator](https://docs.laminas.dev/laminas-validator/validators/undisclosed-password/). This can be used to check new passwords against the Pwned Passwords API from Troy Hunt.

**Comment by Koen1999** [comment]
It's amazing that they actually added this feature!

**Comment by tomudding** [comment]
I have talked with the CBC about this and we have devised a better solution than checking the actual API of Pwned Passwords. As there is plenty of RAM too spare, we can implement an actual Bloom Filter in Redis such that we can check the passwords locally.

We do not even have to generate the bloom filter ourselves, as we can get a copy from [here](https://scotthelme.co.uk/re-bloom-pwned-passwords-v8/).

So to summarise:
1) New Docker container with Redis (and `redisbloom`).
2) Permanently storing the bloom filter dump and load it every time the container is restarted.
3) Query passwords via [palicao/phpRebloom](https://github.com/palicao/phpRebloom) in gewisweb.

```php
use Palicao\PhpRebloom\BloomFilter;
use Palicao\PhpRebloom\RedisClient;
use Palicao\PhpRebloom\RedisConnectionParams;

$bloomFilter = new BloomFilter(
	new RedisClient(
		new Redis(),
		new RedisConnectionParams('redis', 6379),
	)
);

$exists = $bloomFilter->exists('pwned-bloom', strtoupper(sha1($password)));
```
